2006Scape Client/src/main/java/Game.java

```diff
@@ -5587,7 +5590,51 @@ public void drawSplitpublicChat() {
 		}
 
 	}
-
+	public void screenshot(boolean sendMessage, String... subfolders) {
+		try {
+			Window window = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusedWindow();
+			if (window == null) {
+				return;
+			}
+			Point point = window.getLocationOnScreen();
+			int x = (int) point.getX();
+			int y = (int) point.getY();
+			int w = window.getWidth();
+			int h = window.getHeight();
+			Robot robot = new Robot(window.getGraphicsConfiguration().getDevice());
+			Rectangle captureSize = new Rectangle(x, y, w, h);
+			BufferedImage bufferedimage = robot.createScreenCapture(captureSize);
+	
+			// Format the current date and time
+			SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy_MM_dd-HH_mm_ss");
+			String dateTime = dateFormat.format(new Date());
+	
+			// Update the file path and naming
+			String fileExtension = myUsername != null && !myUsername.isEmpty() ? myUsername : "2006Scape";
+			
+			String subfolderPath = String.join(File.separator, subfolders);
+			if (!subfolderPath.isEmpty()) {
+				subfolderPath += File.separator;
+			}
+			
+			String screenshotDir = System.getProperty("user.home") + File.separatorChar + "2006Scape" + File.separatorChar + "screenshots" + File.separatorChar + subfolderPath;
```

Comment:
we should be using the servername var from clientsettings here

Reply:
Wow yep, nice catch